### PR TITLE
Enable lint-staged and add eslint and clippy

### PR DESCRIPTION
**Pull request by wjthieme**
* Enable lint-staged
* Add clippy & eslint
* Add gh actions stript to run linter

### CODE REVIEW

**Review by yugure-orca — DISMISSED** [review]

File: programs/whirlpool/src/util/mod.rs
Comment (yugure-orca) [inline programs/whirlpool/src/util/mod.rs:1]: hmm, clippy is so aggressive.

**Review by yugure-orca — APPROVED** [review]
LGTM with nit

File: programs/whirlpool/src/instructions/swap.rs [hunk 1]
```diff
@@ -86,12 +86,10 @@ pub fn handler(
...
-        if (a_to_b && other_amount_threshold < swap_update.amount_a)
-            || (!a_to_b && other_amount_threshold < swap_update.amount_b)
-        {
-            return Err(ErrorCode::AmountInAboveMaximum.into());
-        }
+    } else if (a_to_b && other_amount_threshold < swap_update.amount_a)
+        || (!a_to_b && other_amount_threshold < swap_update.amount_b)
+    {
+        return Err(ErrorCode::AmountInAboveMaximum.into());
     }
```
[the remark this answers is not in the document]
Comment (yugure-orca) [reply programs/whirlpool/src/instructions/swap.rs:93]: (of course it is logically equivalent and correct)

[the remark this answers is not in the document]
Comment (wjthieme) [reply programs/whirlpool/src/instructions/swap.rs:93]: Yep can add a clippy exception here

File: programs/whirlpool/src/util/mod.rs
Comment (wjthieme) [reply programs/whirlpool/src/util/mod.rs:1]: Yeah and it is also not as configurable as eslint for example. Most rules are either on or off without much you can change. Same goes for 'fmt' I would say

**Comment by wjthieme** [comment]
Test failures are unrelated so will continue with merging

**Pull request merged**